subprojects/launcher/src/test/groovy/org/gradle/tooling/internal/provider/FileSystemWatchingBuildActionRunnerTest.groovy

```diff
@@ -45,28 +47,33 @@ class FileSystemWatchingBuildActionRunnerTest extends Specification {
     def delegate = Mock(BuildActionRunner)
     def buildAction = Mock(BuildAction)
 
-    def "watching virtual file system is informed about watching the file system being #watchFsEnabledString (verbose logging: #verboseLogging)"() {
+    def "watching virtual file system is informed about watching the file system being #watchFsEnabledString (VFS logging: #verboseLogging, watch logging: #debugLogging)"() {
         _ * startParameter.getSystemPropertiesArgs() >> [:]
         _ * startParameter.isWatchFileSystem() >> watchFsEnabled
-        _ * startParameter.isWatchFileSystemVerboseLogging() >> verboseLogging
+        _ * startParameter.isWatchFileSystemDebugLogging() >> watchLogging
+        _ * startParameter.isVfsVerboseLogging() >> vfsLogging
 
         def runner = new FileSystemWatchingBuildActionRunner(delegate)
 
         when:
         runner.run(buildAction, buildController)
         then:
-        1 * watchingHandler.afterBuildStarted(watchFsEnabled, verboseLogging, buildOperationRunner)
+        1 * watchingHandler.afterBuildStarted(watchFsEnabled, vfsLogging, watchLogging, buildOperationRunner)
 
         then:
         1 * delegate.run(buildAction, buildController)
 
         then:
-        1 * watchingHandler.beforeBuildFinished(watchFsEnabled, verboseLogging, buildOperationRunner, _)
+        1 * watchingHandler.beforeBuildFinished(watchFsEnabled, vfsLogging, watchLogging, buildOperationRunner, _)
 
         where:
-        watchFsEnabled | watchFsEnabledString | verboseLogging
-        true           | "enabled"            | true
-        true           | "enabled"            | false
-        false          | "disabled"           | false
+        watchFsEnabled | vfsLogging         | watchLogging
+        true           | VfsLogging.VERBOSE | WatchLogging.NORMAL
```

Comment:
Maybe it would be better to use `combinations()` instead of writing down most of the combinations here.

Reply:
FSW disabled with verbose logging doesn't make sense, and the test would fail.